pkg/apis/config/v1alpha2/extensionservice.go

```diff
@@ -0,0 +1,48 @@
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ExtensionService defines the configuration of the external service
+// that an OSM managed mesh integrates with.
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ExtensionService struct {
+	// Object's type metadata.
+	metav1.TypeMeta `json:",inline"`
+
+	// Object's metadata.
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the specification of the extension service.
+	// +optional
+	Spec ExtensionServiceSpec `json:"spec,omitempty"`
+}
+
+// ExtensionServiceSpec defines the specification of the extension service.
+type ExtensionServiceSpec struct {
```

Comment:
It might be useful to check out the various settings for auth extension https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto

ie: breaking up gRPC vs HTTP, adding channel credentials (if allowing non-meshed traffic), adding a `headersToAdd` or something similar for HTTP. Just a thought

Reply:
I am thinking we could introduce protocol specific fields such as `grpc, http, tcp` etc. on the resource for this purpose. Additionally, we would need a `tls` field for TLS based communication with the extension service. 

I'd prefer to not handle this in this PR as I chose to introduce the min required fields to unblock the Access log + OpenTelemetry integration work - which for the first pass will rely on non-TLS based log exporting.

We could iterate on this API to add protocol specific and TLS fields like you suggested, as they would merely be field additions to the same API version.